### PR TITLE
feat(*): replace values that change with each release with template values

### LINKED ISSUE #226: Write a script to batch-update data in manifests for a given chart
Things that the script should be able to update:
- Docker image tags
- Names of resources
- `imagePullPolicy`s
- Environment variable values

### PR DESCRIPTION
**Pull request by arschles**
This PR is a backport of #224. It removes the need for a script to change the manifests. Instead, it takes advantage of helm templates to change values in the templates that change with each release.

@technosophos @sgoings thoughts?

Fixes #226 because this PR removes the need for a script to change the manifests. Instead, it takes advantage of helm templates to change values in the templates that change with each release.

If the changes in this patch are accepted, similar changes should be made to all other charts, and after this PR is merged, https://github.com/deis/charts/pull/224 should be rebased and changes should be made to the `workflow-beta2` and `workflow-beta2-e2e` charts in there.

Charts to update & test:
- [x] `deis-logger`
- [x] `deis-logger-beta1`
- [x] `router-beta1`
- [x] `router-dev`
- [x] `workflow-beta1`
- [x] `workflow-beta1-e2e`
- [x] `workflow-dev`
- [x] `workflow-dev-e2e`

Other tasks
- [ ] https://github.com/sgoings/chart-mate/issues/56

Also, after this PR is merged, there is an [accompanying docs PR](https://github.com/deis/workflow/pull/169) that should be merged.
Fixes #226

### CODE REVIEW

**Comment by technosophos** [comment]
Another option would be to use the Helm template to just get the pull policy either from a value or from the environment, e.g. `imagePullPolicy: {{env "IMG_PULL_POLICY" | default "alwaysPull"}}`

(Just a suggestion -- it was on the top of my mind as I read this.)

**Comment by slack** [comment]
I like this approach as it dovetails with the thinking in https://github.com/deis/charts/issues/199

This would allow operators to parameterize the install and enable "backing up" an instance of workflow's configuration by hauling around a filled out TOML file. Probably don't need to go that far for beta2 release as it is a goal for beta3.

**Comment by arschles** [comment]
> Probably don't need to go that far for beta2 release as it is a goal for beta3

@slack funny you should mention that, because we're partly the way there in beta2 (parameterizing just the release values) in https://github.com/deis/charts/pull/224

File: workflow-beta1/manifests/deis-minio-rc.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ spec:
       serviceAccount: deis-minio
       containers:
         - name: deis-minio
-          image: quay.io/deis/minio:v2.0.0-beta1
-          imagePullPolicy: IfNotPresent
+          image: quay.io/{{.minio.org}}/minio:{{.minio.docker-tag}}
```
Comment (kmala) [inline workflow-beta1/manifests/deis-minio-rc.yaml]: Did you test this because last when i did helm template failed when it saw a '-' character

**Comment by kmala** [comment]
Where are we calling the helm generate to template the values from the release.toml

**Comment by arschles** [comment]
@kmala good catch. those are incoming - I've made similar changes for the `beta2` release at https://github.com/deis/charts/pull/224. Going to mark this in progress for now until that PR is done

**Comment by arschles** [comment]
I'm moving this to beta3

**Comment by arschles** [comment]
this change may require changes in `bumpver` from chart-mate. cc/ @vdice

**Comment by kmala** [comment]
@vdice this PR might require changes on the Jenkins side on how to update the image when a PR gets merged

**Comment by vdice** [comment]
Thanks @kmala -- we were thinking https://github.com/sgoings/chart-mate/issues/56 would represent the needed change.

File: workflow-dev/tpl/generate_params.toml [hunk 2]
```diff
@@ -47,3 +49,58 @@ key_json = '''Paste JSON data here.'''
 registry_bucket = "your-regsitry-bucket-name"
```
Comment (vdice) [inline workflow-dev/tpl/generate_params.toml]: pre-existing typo: `s/regsitry/registry`

Comment (arschles) [inline workflow-dev/tpl/generate_params.toml]: Fixed in https://github.com/deis/charts/pull/228/commits/674224f2f7e54b7ab695f014a63924d1ac673f16

**Comment by jchauncey** [comment]
If we use `canary` in workflow-dev there is no reason to have chart mate update the toml file.

**Comment by jchauncey** [comment]
If we merge this it should turn the tests back green

**Comment by jchauncey** [comment]
@arschles merge this!

**Pull request merged**